packages/nimble-components/src/tree-view/index.ts

```diff
@@ -70,6 +76,26 @@ export class TreeView extends FoundationTreeView {
         const treeItemChild = item.querySelector('[role="treeitem"]');
         return treeItemChild !== null;
     }
+
+    private selectedItemsChanged(): void {
+        for (let i = 0; i < this.childElementCount; i++) {
+            const item = this.children[i]!;
+            if (item instanceof TreeItem) {
+                item.groupSelected = false;
+            }
+        }
+
+        for (let item of this.selectedItems) {
+            while (item.parentElement !== null && item.parentElement !== this) {
+                item = item.parentElement;
+            }
+            if (item instanceof TreeItem) {
+                item.groupSelected = true;
+            }
+        }
+
+        this.$emit('group-selected-update');
```

Comment:
I'm not sure if we're okay with adding an event just for testing's sake. It's not something I think would be useful for users, so we shouldn't document it. I looked for other ways to have tests wait for this asynchronous update, but this was the best I could think of. Polling a value up to a timeout is also an option, but if you want to verify that a value _doesn't_ change, then you have to wait for the timeout. It could lead to flaky tests if the timeout is chosen poorly.

Reply:
Sorry, I may not be understanding something. Why isn't something like `waitForUpdatesAsync` sufficient for your needs? At worst, I think it would be two calls to it for it to "settle out", and you could add a comment in the test for what each call was expected to accomplish. So, the first call is to wait for the `selected` state to be toggled on various elements, and the second call is waiting for the `selectedItems` to be added to the DOM (which I think will result in `selectedItemsChanged` completing before the second `waitForUpdatesAsync` call is resolved.

Reply:
I'm not sure what each call to `waitForUpdatesAsync` does, e.g. why doesn't everything settle out in the first call? For that reason, I wasn't sure if running it twice was a reliable way to get tests to pass, or if it might introduce flakiness. But it sounds like I need one call to reflect the property value back to the attribute (if `selected` was modified, or the element was clicked), then another call for the `children` directive to update `selectedItems`. I'll make changes to rely on that instead of a new event.